lib/interpreters/periodic_visits.rb

```diff
@@ -371,36 +373,63 @@ def associate_skills(new_vehicle, vehicle_day_index)
       end
     end
 
-    def compute_possible_days(vrp)
-      # for each of this service's visits, computes first and last possible day to be assigned
-      vrp.services.each{ |service|
-        day = @schedule_start
-        nb_services_seen = 0
-
-        # first possible day
-        while day <= @schedule_end && nb_services_seen < service.visits_number
-          if service.unavailable_days.include?(day) || vrp.vehicles.none?{ |v| v.available_at(day) }
-            day += 1
+    def possible_days_from_possible_day(service, possible_days, vehicles, lapse, options)
```

Comment:
I don't really get the purpose of the function. Does it compute for each visit the interval of possible days where it can be performed ?

The following suggestions are mostly renaming proposition to make it more clear and consistent with the naming of each concept/object. (If I correctly understand their purpose)
```suggestion
    def visits_possible_days(service, possible_days, vehicles, lapse, options)
```